File: app/src/main/java/trunk/doi/base/ui/fragment/main/mvp/MainPresenter.java
package trunk.doi.base.ui.fragment.main.mvp;


import com.base.lib.di.scope.FragmentScope;
import com.base.lib.mvp.BasePresenter;

import javax.inject.Inject;


@FragmentScope
public class MainPresenter extends BasePresenter<MainContract.Model, MainContract.View> {

    @Inject
    public MainPresenter(MainContract.Model model, MainContract.View rootView) {
        super(model, rootView);
    }


    @Override
    public void onDestroy() {
        super.onDestroy();

    }
}

File: app/src/main/java/com/gas/app/ui/fragment/main/mvp/MainPresenter.kt
package com.gas.app.ui.fragment.main.mvp

import com.base.lib.di.scope.FragmentScope
import com.base.lib.mvp.BasePresenter
import javax.inject.Inject

@FragmentScope
class MainPresenter @Inject constructor(model: MainContract.Model, rootView: MainContract.View) : BasePresenter<MainContract.Model, MainContract.View>(model, rootView) {
    override fun onDestroy() {
        super.onDestroy()
    }
}